### ISSUE TITLE
AttributeError: 'binance' object has no attribute 'parse_time_frame'

### ISSUE THREAD

**Issue by amustapha** [coding-kitties/investing-algorithm-framework#277]
Amazing work you've put together here, going through the documentation this seems quite light and straight to the point allowing one to build from the ground up. However I've had a bit of trouble setting this up, first seems the [documentation ](https://investing-algorithm-framework.com/) is out of  sync and i did have a to a bit of improvising. However, I couldn't seem to be able to connect to binance or bybit.

```
Cell In[46], line 8
      1 ticker_btc = CCXTOHLCVMarketDataSource(
      2     identifier="BTC-candles",
      3     market="BINANCE",
      4     symbol="BTC/USDT",
      5     time_frame='1h'
      6 )
----> 8 ticker_btc.get_data(
      9     start_date=datetime(2024, 1, 1),
     10     end_date=datetime(2024, 1, 4),
     11 )

File ~/Work/ofalos/crypto-engine/venv/lib/python3.11/site-packages/investing_algorithm_framework/infrastructure/models/market_data_sources/ccxt.py:515, in CCXTOHLCVMarketDataSource.get_data(self, **kwargs)
    504     data = self._get_data_from_storage(
    505         storage_path=storage_path,
    506         symbol=self.symbol,
   (...)
    510         market=self.market
    511     )
    513 if data is None:
    514     # Get the OHLCV data from the ccxt market service
--> 515     data = market_service.get_ohlcv(
    516         symbol=self.symbol,
    517         time_frame=self.time_frame,
    518         from_timestamp=start_date,
    519         to_timestamp=end_date,
    520         market=self.market
    521     )
    523 # if storage path is set, write the data to the storage path
    524 if storage_path is not None:

File ~/Work/ofalos/crypto-engine/venv/lib/python3.11/site-packages/investing_algorithm_framework/infrastructure/services/market_service/ccxt_market_service.py:405, in CCXTMarketService.get_ohlcv(self, symbol, time_frame, from_timestamp, market, to_timestamp)
    401 ohlcv = exchange.fetch_ohlcv(symbol, time_frame, from_time_stamp)
    403 if len(ohlcv) > 0:
    404     from_time_stamp = \
--> 405         ohlcv[-1][0] + exchange.parse_time_frame(time_frame) * 1000
    406 else:
    407     from_time_stamp = to_timestamp

AttributeError: 'binance' object has no attribute 'parse_time_frame'
```

Here's my code
```
ticker_btc = CCXTOHLCVMarketDataSource(
    identifier="BTC-candles",
    market="BINANCE",
    symbol="BTC/USDT",
    time_frame='1h'
)

ticker_btc.get_data(
    start_date=datetime(2024, 1, 1),
    end_date=datetime(2024, 1, 4),
)
```

Is there something I'm doing wrong?

**Comment by MDUYN** [comment]
I am aware of that issue, and have a fix scheduled for coming Friday. Sorry for the inconvenience

**Comment by MDUYN** [comment]
Fixed as part of release v3.8.1: https://github.com/coding-kitties/investing-algorithm-framework/releases/tag/v3.8.1 

Everything should now work with: pip install investing-algorithm-framework==3.8.1
